subprojects/core/src/integTest/groovy/org/gradle/api/tasks/IncrementalTasksIntegrationTest.groovy

```diff
@@ -108,4 +112,70 @@ class FailingIncrementalTasksIntegrationTest extends AbstractIntegrationSpec {
         "remove"      | false       | "with removed outputs is fully rebuilt"
         "none"        | true        | "with unmodified outputs is executed as incremental"
     }
+
+    @Issue("https://github.com/gradle/gradle/issues/9320")
+    def "incremental task with NAME_ONLY input (matching file names and content) detects changed input"() {
+        def taskName = "incrementalTask"
+        def inputs = ('a'..'z').collect { file("${it}/input.txt").createFile() }
+        def modifiedInput = inputs.toList().sort { random() }.first()
+
+        @Language("Groovy") script = """
+            class IncrementalTask extends DefaultTask {
+                @InputFiles
+                @PathSensitive(PathSensitivity.NAME_ONLY)
+                FileCollection inputFiles = project.files(${inputs.collect { "'${it}'" }.join(", ")})
```

Comment:
What will this use? The relative path of the file? The absolute path of the file? There may be problems on Windows, since the `\` won't be escaped.

How about you use `('a'..'z').collect { "'${it}/input.txt'" }.join(", ")`?

Reply:
`Inputs` already contains the files, here I'm just converting it to a Groovy String List.
It's basically the same as what you're recommending.

Reply:
Indeed, fixed.